### PR TITLE
[WIP]Add Func: npugraph_batch_size auto-adjust to different model

### PR DESCRIPTION
**Pull request by chris668899**
### What this PR does / why we need it?
This PR add new function of : npugraph_batch_size can dynamic adjust to different model; before this PR, the npugraph_batch_sizes given from vllm to vllm-ascend always too large, and that may result in ERROR while running on different, with the information: "The resources are insufficient".
Now, with this PR, the code can dynamic adjust npugraph_batch_sizes depend on the model hidden_layer_nums and parallel config, for example:
a. for Qwen2.5-7B, the npugraph_batch_size length is 33 total;
b. for Qwen2.5-72B, the npugraph_batch_size length is 11 total;

### CODE REVIEW

File: vllm_ascend/worker/model_runner_v1.py [hunk 1]
```diff
@@ -968,3 +972,63 @@ def capture_model(self) -> None:
...
+
+    def get_max_support_len(self, max_capture_size, num_hidden_layers) -> int:
+        parallel_type_cnt = 0
+        dp_size = self.vllm_config.parallel_config.data_parallel_size
+        tp_size = self.vllm_config.parallel_config.tensor_parallel_size
+        if dp_size > 1:
+            parallel_type_cnt += 1
+        if tp_size > 1:
+            parallel_type_cnt += 1
+        max_support_len_npugraph = math.floor(max_capture_size /
```
Comment (ganyi1996ppo) [inline vllm_ascend/worker/model_runner_v1.py:1007]: So, bigger model tends to have smaller aclgraph cache size right? what if bigger model also bounds to the host?

File: vllm_ascend/worker/model_runner_v1.py [hunk 2]
```diff
@@ -968,3 +972,63 @@ def capture_model(self) -> None:
...
+            logger.info(
+                "Model:%s-num_hidden_layers:%d no need adjust npugraph_bash_size, list_len: %s",
+                self.vllm_config.model_config.architectures[0],
+                num_hidden_layers, len(original_npugraph_batch_sizes))
+
+    def get_max_support_len(self, max_capture_size, num_hidden_layers) -> int:
+        parallel_type_cnt = 0
+        dp_size = self.vllm_config.parallel_config.data_parallel_size
+        tp_size = self.vllm_config.parallel_config.tensor_parallel_size
+        if dp_size > 1:
```
Comment (ganyi1996ppo) [inline vllm_ascend/worker/model_runner_v1.py:1003]: So the bigger the parallel size, the smaller the graph step? Should be bigger right?

Comment (chris668899) [reply vllm_ascend/worker/model_runner_v1.py:1003]: The types of parallel strategies influence the length of the list. Therefore, the more types of parallel strategies there are, the shorter the list becomes. However, the maximum supported batch_size value in the list remains unchanged.

File: tests/singlecard/test_dynamic_npugraph_batchsize.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+# limitations under the License.
+# This file is a part of the vllm-ascend project.
+#
+import pytest
+import torch
+from torch_npu.op_plugin.atb._atb_ops import _register_atb_extensions
+from vllm import LLM, SamplingParams
+
+_register_atb_extensions()
+torch.cuda.CUDAGraph = torch.npu.NPUGraph
```
Comment (MengqingCao) [inline tests/singlecard/test_dynamic_npugraph_batchsize.py]:
```suggestion
TODO: revert me when cuda hard code is fixed in `VllmBackend`
torch.cuda.CUDAGraph = torch.npu.NPUGraph
```

File: tests/singlecard/test_dynamic_npugraph_batchsize.py [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# This file is a part of the vllm-ascend project.
+#
+import pytest
+import torch
+from torch_npu.op_plugin.atb._atb_ops import _register_atb_extensions
+from vllm import LLM, SamplingParams
+
+_register_atb_extensions()
```
Comment (MengqingCao) [inline tests/singlecard/test_dynamic_npugraph_batchsize.py]: what does this do?

File: tests/singlecard/test_dynamic_npugraph_batchsize.py [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+torch.cuda.CUDAGraph = torch.npu.NPUGraph
+
+MODELS = [
+    "Qwen/Qwen2.5-0.5B-Instruct",
+]
+
+TENSOR_PARALLELS = [2]
```
Comment (MengqingCao) [inline tests/singlecard/test_dynamic_npugraph_batchsize.py]: This is a multicard ut, let's move this to path `tests/multicard` to make sure it is tested as expected

Comment (chris668899) [reply tests/singlecard/test_dynamic_npugraph_batchsize.py]: has been moved to multicard

File: tests/singlecard/test_dynamic_npugraph_batchsize.py (hunk 3 above)
Comment (chris668899) [reply tests/singlecard/test_dynamic_npugraph_batchsize.py]: ok

**Comment by ganyi1996ppo** [comment]
Please don't merge this PR, we may still need to discuss it with torch_npu and CANN team on this. This solution is neither follow the cuda behavior nor good for performance.

File: tests/singlecard/test_dynamic_npugraph_batchsize.py (hunk 4 above)
Comment (ganyi1996ppo) [reply tests/singlecard/test_dynamic_npugraph_batchsize.py]: torch_npu needs to preload atb's .so before the dyanmo trace procedure.

Comment (chris668899) [reply tests/singlecard/test_dynamic_npugraph_batchsize.py]: "_register_atb_extensions()" has been removed

**Comment by ganyi1996ppo** [comment]
Please replace all the npugraph to aclgraph.

**Comment by ganyi1996ppo** [comment]
For now, seems we don't have much choice on this, for the large model with lots of layers and comm group, we may only have small number of aclgraph cached in memory. Which means enormous padding may happened in many scenario and thus cause the performance regression.  cc @wangxiyuan @Yikun

**Comment by chris668899** [comment]
note: original PR is : https://github.com/vllm-project/vllm-ascend/pull/739, now change to this PR.
